import_token.go

```diff
@@ -94,16 +95,62 @@ func (c *Client) GetImportTokenTransportKey(ctx context.Context) (*ImportTokenKe
 }
 
 // EncryptNonce will wrap the KP generated nonce with the users key-material
-func EncryptNonce(key, value, iv string) (string, string, error) {
-	return encryptNonce(key, value, iv)
+func EncryptNonce(key, value, iv, alg string) (string, string, error) {
+	if alg == "CBC" {
+		return encryptNonceWithCBC(key, value, iv)
+	}
+
+	return encryptNonceWithGCM(key, value, iv)
 }
 
 // EncryptKey will encrypt the user key-material with the public key from key protect
 func EncryptKey(key, pubkey string) (string, error) {
 	return encryptKey(key, pubkey)
 }
 
-func encryptNonce(key, value, iv string) (string, string, error) {
+func encryptNonceWithCBC(key, value, iv string) (string, string, error) {
+	keyMat, err := base64.StdEncoding.DecodeString(key)
+	if err != nil {
+		panic(err)
+	}
+
+	nonce, err := base64.StdEncoding.DecodeString(value)
+	if err != nil {
+		panic(err)
+	}
+
+	block, err := aes.NewCipher(keyMat)
+	if err != nil {
+		panic(err)
+	}
+
+	// PKCS7 Padding
+	paddingLength := aes.BlockSize - len(nonce)%aes.BlockSize
+	paddingBytes := []byte{byte(paddingLength)}
+	paddingText := bytes.Repeat(paddingBytes, paddingLength)
+	nonce = append(nonce, paddingText...)
+
+	var newIv []byte
+	if iv != "" {
+		newIv = []byte(iv)
+	} else {
+		newIv = make([]byte, aes.BlockSize)
+	}
+
+	// Generate an IV to achieve semantic security
+	if _, err := io.ReadFull(rand.Reader, newIv); err != nil {
+		panic(err)
+	}
+
+	cipherText := make([]byte, len(nonce))
+
+	mode := cipher.NewCBCEncrypter(block, newIv)
+	mode.CryptBlocks(cipherText, nonce)
+
+	return base64.StdEncoding.EncodeToString(cipherText), base64.StdEncoding.EncodeToString(newIv), nil
+}
+
+func encryptNonceWithGCM(key, value, iv string) (string, string, error) {
```

Comment:
Yeah no... you can't just go and change the function signiture to what you like. You can add a new function tho.